src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4465,193 @@ subroutine cat_enkf_increments(                               &
+       
+       N_select_varnames  = 1      
+       
+       select_varnames(1) = 'asnow'
+       
+       call get_select_species(                                           &
+            N_select_varnames, select_varnames(1:N_select_varnames),      &
+            N_obs_param, obs_param, N_select_species, select_species )
+       
+       allocate(select_tilenum(1))
+       
+       swe_incr = 0.   ! total SWE increment; initialize to NO CHANGE
+       
+       ! loop through tiles and compute increments
+       
+       do kk=1,N_catd
+          
+          ! find observations for tile kk
+          
+          select_tilenum(1) = l2f(kk)
+          
+          call get_ind_obs(                                           &
+               N_obs,            Observations,                        &
+               1,                select_tilenum,                      &
+               N_select_species, select_species(1:N_select_species),  &
+               N_selected_obs,   ind_obs )
+          
+          if (N_selected_obs > 0) then
+             
+             ! NEED TO ALLOW FOR MULTIPLE asnow OBS (I.E., FROM MULTIPLE SENSORS, SAY, MODIS AND VIIRS)
+             !
+             ! if statement added to maintain 0-diff for now; can be removed later
+             ! (that is, compute average even when N_selected_obs=1)
+             
+             if (N_selected_obs > 1) then    
+                
+                ! compute average obs value
+                
+                tmp_obs = sum(Observations(ind_obs(1:N_selected_obs))%obs)
+                
+                tmp_obs = tmp_obs/real(N_selected_obs)
+                
+             else
+                
+                tmp_obs = Observations(ind_obs(1))%obs
+                
+             end if
+             
+             
+             do n_e=1,N_ens  ! compute analysis separately for each ensemble member
+                
+                ! 1. Get model forecast snow cover area fraction and total SWE
+                
+                asnow_fcst  = asnow(kk,n_e)
+                swe_fcst    = sum(cat_progn(kk,n_e)%wesn(1:N_snow))
+                
+                ! 2. Calculate SWE increment based on Toure et al (2018) eq 1
+                
+                if     (asnow_fcst .lt. tmp_obs * SCF_ANA_ALPHA) then
+                   
+                   ! ADD SNOW:    Forecast SCF is less than observed SCF (after "bias" adjustment with alpha)
+                   
+                   swe_incr(kk,n_e) = SCF_ANA_MAXINCRSWE * (tmp_obs - asnow_fcst/SCF_ANA_ALPHA)
+                   
+                elseif (tmp_obs .lt. SCF_ANA_BETA) then
+                   
+                   ! REMOVE SNOW: Simulated SCF is greater than observed SCF (after "bias" adjustment)
+                   !              and observed SCF is less than beta threshold
+                   
+                   swe_incr(kk,n_e) = (-1.) *  SCF_ANA_MAXINCRSWE * asnow_fcst * (1. - tmp_obs/SCF_ANA_BETA)
```

Comment:
I noticed that there's a minor blemish with the revised version of Toure et al 2018 here.  Specifically, swe_incr has a small discontinuity across the line that separates the negative and positive increments (line through origin with slope alpha).  I don't think this necessarily needs to be changed, but we may want to discuss.  To illustrate, here's a plot with the parameters alpha=0.6, beta=0.55, and maxsweincr=5.0 (as currently hardwired in clsm_ensupd_glob_param.F90):
![tmpToureEtAl2018eq1](https://github.com/GEOS-ESM/GEOSldas/assets/54944691/c2dff88e-36d3-49f3-a262-348ac8826b57)